src/main/java/com/webservices/company/repository/CompanyRepository.java

```diff
@@ -1,5 +1,97 @@
+import org.springframework.jdbc.core.PreparedStatementCreator;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.stereotype.Component;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.Timestamp;
+import java.util.List;
+
+@Component
+public class CompanyRepository implements ICompanyRepository {
+
+    private JdbcTemplate jdbcTemplate;
+    private RowMapper<Company> rowMapper;
+
+    public CompanyRepository(JdbcTemplate jdbcTemplate, RowMapper<Company> rowMapper) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.rowMapper = rowMapper;
+
+    }
+
+    @Override
+    public Company add(Company companyToBeAdded) {
+        // Preparing the insert statement
+        PreparedStatementCreator preparedStatementCreator = (Connection connection) -> {
+            PreparedStatement preparedStatement = connection.prepareStatement("INSERT INTO public.companies(\n" +
+                    "\tname, address, number_of_employees, \"dateFound\", type_of_business)\n" +
+                    "\tVALUES (? ,?, ?, ?, ?)", new String[]{"id"}); // first index is 1
+            preparedStatement.setString(1, companyToBeAdded.getName());
+            preparedStatement.setString(2, companyToBeAdded.getAddress());
+            preparedStatement.setInt(3, companyToBeAdded.getNumberOfEmployees());
+            preparedStatement.setTimestamp(4, Timestamp.from(companyToBeAdded.getDateFound()));
+            preparedStatement.setString(5, companyToBeAdded.getTypeOfBusiness());
+            return preparedStatement;
+        };
+        GeneratedKeyHolder generatedKeyHolder = new GeneratedKeyHolder();
+        // Actual insert happens when calling update method
+        jdbcTemplate.update(preparedStatementCreator, generatedKeyHolder);
+        Number keyGenerated = generatedKeyHolder.getKey();
+        Long key = keyGenerated.longValue();
+        return get(key);
+    }
+
+    @Override
+    public Company get(Long id) {
+        Company company = jdbcTemplate.queryForObject("select * from companies where id = " + id, rowMapper);
+        return company;
+    }
+
+    @Override
+    public Company update(Long id, Company updatedCompany) {
+        // Preparing the update statement
+        PreparedStatementCreator preparedStatementCreator = (Connection connection) -> {
+            PreparedStatement preparedStatement = connection.prepareStatement("UPDATE public.companies\n" +
+                    "\tSET name=?, address=?, number_of_employees=?, \"dateFound\"=?, type_of_business=?\n" +
+                    "\tWHERE id = ?");
+            preparedStatement.setString(1, updatedCompany.getName());
+            preparedStatement.setString(2, updatedCompany.getAddress());
+            preparedStatement.setInt(3, updatedCompany.getNumberOfEmployees());
+            preparedStatement.setTimestamp(4, Timestamp.from(updatedCompany.getDateFound()));
```

Comment:
Timestamp.from(updatedCompany.getDateFound()) can throw NPE in case updatedCompany.getDateFound() retruns null.